### PR TITLE
#184557729 Changing order status

### PR DESCRIPTION
**Pull request by fabmukunzi**
### What does this PR do?

Changing order status

### Description of Task to be completed?

A seller should be able to change Product order status and then if all products in the same order are accepted
then order status is changed to accepted and when one of the product order is declined order status become declined
otherwise order is pending

### How should this be manually tested?

1. Clone [repo](https://github.com/atlp-rwanda/vikings-ec-bn.git)
2. Change into the directory **vikings-ec-bn**
3. Checkout to this branch:` feature/change-order-status-184557729`
4. Run the command `npm install` to install packages
5. Run` npm run reset:db`
6. Run `npm run dev`
7. In postman create a patch request **localhost:5000/api/v1/sales/{salesId}/status**

### What are the relevant pivotal tracker stories?

[#184557729](https://www.pivotaltracker.com/story/show/184557729)

### CODE REVIEW

**Review by nshimiyejayd — APPROVED** [review]

**Review by irakozetony — APPROVED** [review]

File: src/middlewares/sales.middleware.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Sales } from '../database/models/index';
+
+export const checkIfProductOrderExistsById = async (req, res, next) => {
+  const { salesId } = req.params;
+  const {id}=req.user
```
Comment (Aline-Uwera) [inline src/middlewares/sales.middleware.js]: where does this id being used?

**Review by Aline-Uwera — APPROVED** [review]

**Comment by ncutixavier** [comment]
Please fix all checks @fabmukunzi

File: src/controllers/sales.controller.js [hunk 2]
```diff
@@ -1,53 +1,96 @@
...
-            });
-        }
+  static async getOrderSales(req, res) {
+    try {
+      const orderId = req.order.id;
+      const sales = await SalesService.getOrderSales(orderId);
+      let buyerSales = [];
+      for (const sale of sales) {
+        buyerSales.push(sale);
+      }
```
Comment (ncutixavier) [inline src/controllers/sales.controller.js]: Why don't you remove this and return buyerSales?

File: src/services/order.service.js [hunk 3]
```diff
@@ -1,28 +1,32 @@
-import { Orders } from '../database/models/index';
+import { Order } from '../database/models/index';
 export class OrderService {
-	static async getSingleOrder(orderId) {
-		return Orders.findByPk(orderId);
-	}
+  static async getSingleOrder(orderId) {
+	console.log("==============",Order)
```
Comment (ncutixavier) [inline src/services/order.service.js]: remove all logs

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**